backends-clickhouse/src/main/scala/io/glutenproject/execution/CHHashAggregateExecTransformer.scala

```diff
@@ -197,6 +218,15 @@ case class CHHashAggregateExecTransformer(
       })
     // Get the aggregate function nodes.
     val aggregateFunctionList = new util.ArrayList[AggregateFunctionNode]()
+
+    val distinct_modes = aggregateExpressions.map(_.mode).distinct
+    // quick check
+    if (distinct_modes.contains(PartialMerge)) {
+      if (distinct_modes.contains(Final)) {
+        throw new IllegalStateException("PartialMerge co-exists with Final")
```

Comment:
A quick question, is this a case found in unit test?

Reply:
No. it's just a guard to prevent unsupported cases